### ISSUE TITLE
Eclipse Che - Create New Workspace Issue from Che Dashboard

### ISSUE THREAD

**Issue by csa-nathan** [eclipse/che#21163]
### Summary

Have successfully installed eclipse Che in AKS cluster and running without any errors. 

Chectl version 7.40.0 and able to access Che Dashboard, Keycloak, devfile-registry, plugin-registry URLs.

Have logged in to eclipse Dashboard and trying to create workspace form the already defined samples. 

On creation of new workspace by user, in the log it says
Pulling image "quay.io/eclipse/che-machine exec@sha256:2e244dd5b3ed0baf00570a21a904b4066a35b1618a0a0fb0aa02e345c4faeeeb"

Then after couple of minutes get an error,  Failed to run the workspace: "Server 'terminal' in container 'che-machine-execk63' not available."

Any sample workspace provided in the Dashboard, Chosen to create either gives an error

Failed to run the workspace: "Server 'terminal' in container 'che-machine-execk63' not available.

OR

Failed to run the workspace: "Server 'theia' in container 'theia-idepok' not available." 

Is it a configuration issue or a BUG as part of Eclipse Che Server 7.40.0 ( Using Chectl Version 7.40.0)?

Not able to  identify the problem and resolve this issue for quite some time.

Looking for help to resolve this issue.

**Comment by csa-nathan** [comment]
![image](https://user-images.githubusercontent.com/99651183/153859693-cf87fc2d-4dfe-4f9a-8334-ad3eeb3f562c.png)

**Comment by csa-nathan** [comment]
For better understanding of the issue, sharing the screen grab of the error that gets displayed.

**Comment by tolusha** [comment]
Hello. @csa-nathan 

1. Try to patch checluster CR to enable debug logging
```
kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_LOGGER_CONFIG\": \"che.infra.request-logging=TRACE,io.fabric8.kubernetes.client=DEBUG,org.eclipse.che.api.workspace.server.hc.HttpConnectionServerChecker=DEBUG\"}}}}" --type=merge -n eclipse-che
```
2. wait until che-server pod is restatred and try to create a workspace again
3. grab and attach che-server pod logs
```
kubectl logs deploy/che -n eclipse-che
```

**Comment by csa-nathan** [comment]
Hello Tolusha,

Thanks for the help extended.  

I have enabled the log as per your directions.

Once the error occurs, it removes the namespace admin-che which has 5 containers and that's the final details i get in the log as below.

jQ2MTY0MiwiaWF0IjoxNjQ0OTI1NjQyLCJpc3MiOiJodHRwczovL2VjbGlwc2VpZGUtZWNsaXBzZWNoZXJlc291ci1lNDhiMDUtMjE2ZWUwOWYuaGNwLmNlbnRyYWxpbmRpYS5hem1rOHMuaW8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImVjbGlwc2UtY2hlIiwicG9kIjp7Im5hbWUiOiJjaGUtNzU2OWJmY2Y5Zi0yMnZ0bCIsInVpZCI6Ijk3MWFiYWQzLTBlN2ItNDI5MS1hODg3LTdhYjg4YjI1OWQ4MSJ9LCJzZXJ2aWNlYWNjb3VudCI6eyJuYW1lIjoiY2hlIiwidWlkIjoiZjYzYjE3MjAtYWE0NC00ZmVhLTliY2QtNTJlN2NhMThhOTY0In0sIndhcm5hZnRlciI6MTY0NDkyOTI0OX0sIm5iZiI6MTY0NDkyNTY0Miwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmVjbGlwc2UtY2hlOmNoZSJ9.YxF4O8ci3FVEFfXKyImeT3dmCHSMZfU0QF_fdlsrXnxtlUFGPESjK9PjE1t-El4R4iKzHz6W--EQwS8FpPnZZvR2g-XPgVz4ciSO1SNIWV8LkGTxVAjV0s5farP4_18VotQ2s88gwIw7olLNKKbzUgrzUJrEObHqqj5bmtvlYEBxh0U4dlfrj1jHJdJdpaw8_78XxZxLrCbcHEHL18blFCX-66pVimi0UbXXI-u13kZ8qYmqrOrOzWa868SkCv2M6Bs0IrglSmQqbHpS4cEX4mQhEaegyiBviQoAVU1dXbRKL1bbYsgJS_FNCIW3KxXCkRpkN9T3BZqmf14c_s-TL4RWltBc1gJQMWlOwa9lDD0mtQpPmPg4e460cHmRKJvDO7qJ4_jqsr7_ZyLWeNewrXMnRjCa6cCzWtJRhKRTmf_zXiafvsoS2bbyvqAgDbwgO6dYktA_1oHFbYN2x7LN8CgOZtzXzXWQpXkKM6g_VtARewZtgS8Xp-JeJfdz8uvW6BepEyFWqRVUjylgwP-5jtZQOkS5-MdVlT1EwnESbUIv3w63RkhHhGBizroX2nRor0hgRiCNw4n9VFMlz3JvmM8bxhOpjmfqXLY0cU6fnrjM5Xc71DazcQsUP1vOUoDx1jNPBKFahfQXkSOgZshVZ9PjfCXKlqRCBN2cshKD55c
2022-02-15 12:03:58,771[aceSharedPool-1]  [TRACE] [che.infra.request-logging 210]       -
2022-02-15 12:03:58,771[aceSharedPool-1]  [TRACE] [che.infra.request-logging 212]       - {"apiVersion":"v1","kind":"DeleteOptions","propagationPolicy":"Background"}
2022-02-15 12:03:58,771[aceSharedPool-1]  [TRACE] [che.infra.request-logging 213]       - --> END DELETE (75-byte body)
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 235]       - <-- 200 https://10.0.0.1/api/v1/namespaces/admin-che/configmaps/workspaceioyfowrq20uehi3b.sshconfigmap (9ms)
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - audit-id: ea5c9ba3-3509-417d-806b-7be4f921bf7a
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - cache-control: no-cache, private
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - content-type: application/json
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - x-kubernetes-pf-flowschema-uid: b2d75920-a1a6-40a2-a8ad-5273fbb950bc
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - x-kubernetes-pf-prioritylevel-uid: 79215104-e845-40d7-b61d-c7d01f9e5968
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - content-length: 194
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 302]       - date: Tue, 15 Feb 2022 12:03:58 GMT
2022-02-15 12:03:58,780[aceSharedPool-1]  [TRACE] [che.infra.request-logging 284]       -
2022-02-15 12:03:58,781[aceSharedPool-1]  [TRACE] [che.infra.request-logging 285]       - {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Success","details":{"name":"workspaceioyfowrq20uehi3b.sshconfigmap","kind":"configmaps","uid":"9e32243c-baaf-42bb-a4d2-7cf246d6b6bb"}}

2022-02-15 12:03:58,781[aceSharedPool-1]  [TRACE] [che.infra.request-logging 292]       - <-- END HTTP (194-byte body)
2022-02-15 12:03:58,781[aceSharedPool-1]  [DEBUG] [f.k.c.d.i.AbstractWatchManager 195]  - Force closing the watch io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager@78b6509e
2022-02-15 12:03:58,781[aceSharedPool-1]  [DEBUG] [f.k.c.d.i.AbstractWatchManager 100]  - Ignoring duplicate firing of onClose event
2022-02-15 12:03:58,843[aceSharedPool-1]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 1053]  - Workspace 'admin:golang-e7ek' with id 'workspaceioyfowrq20uehi3b' start failed

In between, i could see the error  

Failed to establish http connection to check server terminal : ....  and 
Failed to establish http connection to check server theia: .....  The details are attached below for your reference.

![EclipseChe-Log1](https://user-images.githubusercontent.com/99651183/154059596-b0fb930b-84c2-4e00-93f8-059b1b83970d.png)


Kindly guide me to resolve this long pending issue.

With Thanks,
Nathan CSA.

**Comment by csa-nathan** [comment]
Complete Log of Che Server Attached for your reference. Sorry, Its a huge log....  
[EclipseCheErrorLog-150220221750.txt](https://github.com/eclipse/che/files/8069510/EclipseCheErrorLog-150220221750.txt)

**Comment by csa-nathan** [comment]
The URLs for Terminal and theia are dynamic ( new pods get created) and trying to check whether its active.   When I installed Eclipse Che, i created A record for Keycloak, dev-file-registry, plugin-registry and for che-eclipse-che.ide. Only then the URLs started working.

But for these dynamic URLs, as there are no entries, it fails. I could sense the problem.  I have installed the Eclipse Che and validated the DNS Challenge ( LetsEncrypt SSL Certificate gets issued, i think with wild card SSL only) using *.ide.banyanpro.uk where banyanpro.uk is the domain. 

How to resolve this issue?  Once any sub-domain URLs are able to validate using a wild card SSL, probably the error might go off and the workspace gets created.

**Comment by csa-nathan** [comment]
> Hello. @csa-nathan
> 
> 1. Try to patch checluster CR to enable debug logging
> 
> ```
> kubectl patch checluster/eclipse-che --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"CHE_LOGGER_CONFIG\": \"che.infra.request-logging=TRACE,io.fabric8.kubernetes.client=DEBUG,org.eclipse.che.api.workspace.server.hc.HttpConnectionServerChecker=DEBUG\"}}}}" --type=merge -n eclipse-che
> ```
> 
> 2. wait until che-server pod is restatred and try to create a workspace again
> 3. grab and attach che-server pod logs
> 
> ```
> kubectl logs deploy/che -n eclipse-che
> ```
Hello Tolusha,

Thanks for the help extended.

I have enabled the log as per your directions.

Once the error occurs, it removes the namespace admin-che which has 5 containers and that's the final details i get in the log as below.

jQ2MTY0MiwiaWF0IjoxNjQ0OTI1NjQyLCJpc3MiOiJodHRwczovL2VjbGlwc2VpZGUtZWNsaXBzZWNoZXJlc291ci1lNDhiMDUtMjE2ZWUwOWYuaGNwLmNlbnRyYWxpbmRpYS5hem1rOHMuaW8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImVjbGlwc2UtY2hlIiwicG9kIjp7Im5hbWUiOiJjaGUtNzU2OWJmY2Y5Zi0yMnZ0bCIsInVpZCI6Ijk3MWFiYWQzLTBlN2ItNDI5MS1hODg3LTdhYjg4YjI1OWQ4MSJ9LCJzZXJ2aWNlYWNjb3VudCI6eyJuYW1lIjoiY2hlIiwidWlkIjoiZjYzYjE3MjAtYWE0NC00ZmVhLTliY2QtNTJlN2NhMThhOTY0In0sIndhcm5hZnRlciI6MTY0NDkyOTI0OX0sIm5iZiI6MTY0NDkyNTY0Miwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmVjbGlwc2UtY2hlOmNoZSJ9.YxF4O8ci3FVEFfXKyImeT3dmCHSMZfU0QF_fdlsrXnxtlUFGPESjK9PjE1t-El4R4iKzHz6W--EQwS8FpPnZZvR2g-XPgVz4ciSO1SNIWV8LkGTxVAjV0s5farP4_18VotQ2s88gwIw7olLNKKbzUgrzUJrEObHqqj5bmtvlYEBxh0U4dlfrj1jHJdJdpaw8_78XxZxLrCbcHEHL18blFCX-66pVimi0UbXXI-u13kZ8qYmqrOrOzWa868SkCv2M6Bs0IrglSmQqbHpS4cEX4mQhEaegyiBviQoAVU1dXbRKL1bbYsgJS_FNCIW3KxXCkRpkN9T3BZqmf14c_s-TL4RWltBc1gJQMWlOwa9lDD0mtQpPmPg4e460cHmRKJvDO7qJ4_jqsr7_ZyLWeNewrXMnRjCa6cCzWtJRhKRTmf_zXiafvsoS2bbyvqAgDbwgO6dYktA_1oHFbYN2x7LN8CgOZtzXzXWQpXkKM6g_VtARewZtgS8Xp-JeJfdz8uvW6BepEyFWqRVUjylgwP-5jtZQOkS5-MdVlT1EwnESbUIv3w63RkhHhGBizroX2nRor0hgRiCNw4n9VFMlz3JvmM8bxhOpjmfqXLY0cU6fnrjM5Xc71DazcQsUP1vOUoDx1jNPBKFahfQXkSOgZshVZ9PjfCXKlqRCBN2cshKD55c
2022-02-15 12:03:58,771[aceSharedPool-1] [TRACE] [che.infra.request-logging 210] -
2022-02-15 12:03:58,771[aceSharedPool-1] [TRACE] [che.infra.request-logging 212] - {"apiVersion":"v1","kind":"DeleteOptions","propagationPolicy":"Background"}
2022-02-15 12:03:58,771[aceSharedPool-1] [TRACE] [che.infra.request-logging 213] - --> END DELETE (75-byte body)
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 235] - <-- 200 https://10.0.0.1/api/v1/namespaces/admin-che/configmaps/workspaceioyfowrq20uehi3b.sshconfigmap[](https://user-images.githubusercontent.com/99651183/154062261-0083c1d3-d747-43df-afd3-c6e511271f07.png) (9ms)
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - audit-id: ea5c9ba3-3509-417d-806b-7be4f921bf7a
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - cache-control: no-cache, private
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - content-type: application/json
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - x-kubernetes-pf-flowschema-uid: b2d75920-a1a6-40a2-a8ad-5273fbb950bc
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - x-kubernetes-pf-prioritylevel-uid: 79215104-e845-40d7-b61d-c7d01f9e5968
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - content-length: 194
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 302] - date: Tue, 15 Feb 2022 12:03:58 GMT
2022-02-15 12:03:58,780[aceSharedPool-1] [TRACE] [che.infra.request-logging 284] -
2022-02-15 12:03:58,781[aceSharedPool-1] [TRACE] [che.infra.request-logging 285] - {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Success","details":{"name":"workspaceioyfowrq20uehi3b.sshconfigmap","kind":"configmaps","uid":"9e32243c-baaf-42bb-a4d2-7cf246d6b6bb"}}

2022-02-15 12:03:58,781[aceSharedPool-1] [TRACE] [che.infra.request-logging 292] - <-- END HTTP (194-byte body)
2022-02-15 12:03:58,781[aceSharedPool-1] [DEBUG] [f.k.c.d.i.AbstractWatchManager 195] - Force closing the watch io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager@78b6509e
2022-02-15 12:03:58,781[aceSharedPool-1] [DEBUG] [f.k.c.d.i.AbstractWatchManager 100] - Ignoring duplicate firing of onClose event
2022-02-15 12:03:58,843[aceSharedPool-1] [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 1053] - Workspace 'admin:golang-e7ek' with id 'workspaceioyfowrq20uehi3b' start failed

In between, i could see the error

Failed to establish http connection to check server terminal : .... and
Failed to establish http connection to check server theia: ..... The details are attached below for your reference.

EclipseChe-Log1

Kindly guide me to resolve this long pending issue.

With Thanks,

**Comment by csa-nathan** [comment]
The URLs for terminal and theia in the error log are dynamic in terms of sub-domain ( new pod with multiple containers get created during workspace creation) and trying to check whether its active. When I installed Eclipse Che, i created "A" record for Keycloak, dev-file-registry, plugin-registry and for che-eclipse-che.ide with my domain registrar. Only then the URLs started working.

But for these dynamic URLs, as there are no entries, it fails. I could sense the problem. I have installed the Eclipse Che and validated the DNS Challenge ( LetsEncrypt SSL Certificate gets issued, i think with wild card SSL only) using *.ide.banyanpro.uk where banyanpro.uk is the domain.

How to resolve this issue? Once any sub-domain URLs are able to validate using a wild card SSL, probably the error might go off and the workspace shall get created.

Kindly help me to resolve this issue...

**Comment by csa-nathan** [comment]
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"workspaceioyfowrq20uehi3b.go-cli-c7c75476c-55tf5\" is forbidden: User \"system:serviceaccount:eclipse-che:che\" cannot get resource \"pods/log\" in API group \"\" in the namespace \"admin-che\"","reason":"Forbidden","details":{"name":"workspaceioyfowrq20uehi3b.go-cli-c7c75476c-55tf5","kind":"pods"},"code":403}


The above seem to be the issue. After this failure, it starts giving the following error

2022-02-15 12:01:04,895[ServersChecker]   [DEBUG] [.h.HttpConnectionServerChecker 116]  - Failed to establish http connection to check server 'terminal:https://serverj1xle2wc-jwtproxy-server-4403.ide.banyanpro.uk/'. Cause: serverj1xle2wc-jwtproxy-server-4403.ide.banyanpro.uk
2022-02-15 12:01:04,916[ServersChecker]   [DEBUG] [.h.HttpConnectionServerChecker 116]  - Failed to establish http connection to check server 'theia:https://serverj1xle2wc-jwtproxy-server-4402.ide.banyanpro.uk/'. Cause: serverj1xle2wc-jwtproxy-server-4402.ide.banyanpro.uk

It tries to establish http connection multiple times, then give up and display the following exception...

workspace workspaceioyfowrq20uehi3b.
org.eclipse.che.api.workspace.server.spi.InfrastructureException: Server 'terminal' in container 'che-machine-exec78x' not available.
	at org.eclipse.che.api.workspace.server.hc.ServerChecker$ServerCheckingTask.run(ServerChecker.java:122)
	at java.base/java.util.TimerThread.mainLoop(Unknown Source)
	at java.base/java.util.TimerThread.run(Unknown Source)
2022-02-15 12:03:53,159[aceSharedPool-1]  [DEBUG] [f.k.c.d.i.AbstractWatchManager 195]  - Force closing the watch io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager@78b6509e
2022-02-15 12:03:53,161[aceSharedPool-1]  [DEBUG] [i.f.kubernetes.client.Watcher 36]    - Watcher closed
2022-02-15 12:03:53,161[aceSharedPool-1]  [DEBUG] [k.c.d.i.WatchConnectionManager 45]   - Closing websocket okhttp3.internal.ws.RealWebSocket@4c423842
2022-02-15 12:03:53,162[://10.0.0.1/...]  [DEBUG] [c.d.i.WatcherWebSocketListener 109]  - Socket closing: 
2022-02-15 12:03:53,162[://10.0.0.1/...]  [DEBUG] [c.d.i.WatcherWebSocketListener 115]  - WebSocket close received. code: 1000, reason: 

Finally, deletes the workspace workspaceioyfowrq20uehi3b.go-cli-c7c75476c-55tf5.


How to resolve this error. Need help...

With Thanks,
Nathan CSA.

**Comment by tolusha** [comment]
To be honest 7.40.0 is quite old version of Eclipse Che.
I don't have any capacity to dig into the problem.
Could you try the latest 7.43.0 ?

**Comment by csa-nathan** [comment]
> To be honest 7.40.0 is quite old version of Eclipse Che. I don't have any capacity to dig into the problem. Could you try the latest 7.43.0 ?

I started with 7.42 and had difficulty in the installation, especially on the OIDC Provider, Keycloak.  Not installing by default, which was clearly highlighted in one of the post.  Have to install it and then configure it, which is again a new thing for me.

I then moved to 7.41 version, which installs Keycloak by default. I started this activity in the first week of Jan and since 7.41 got released towards Dec 30,2021 thought of going with 7.40. I also felt not much of a difference between 7.41.1, 7.41.0 and 7.40 with only weeks difference in release time.

**Comment by csa-nathan** [comment]
Hello Tolusha,

Should I take 7.41.2 version signed off by yourself and try it out?  Will Keycloak get installed by default in this version? 

Spent too much time on version 7.42.0 for Keycloak and found the default installation got removed.

Should I enable devWorkspace to true or leave it to default false? 

With Thanks,
Nathan CSA

**Comment by csa-nathan** [comment]
> Hello Tolusha,
> 
> Should I take 7.41.2 version signed off by yourself and try it out? Will Keycloak get installed by default in this version?
> 
> Spent too much time on version 7.42.0 for Keycloak and found the default installation got removed.
> 
> Should I enable devWorkspace to true or leave it to default false?
> 

Should I take 7.41.2 version signed off by yourself and try it out? Will Keycloak get installed by default in this version?

Spent too much time on version 7.42.0 for Keycloak and found the default installation got removed.

Should I enable devWorkspace to true or leave it to default false?

With Thanks,
Nathan CSA

**Comment by csa-nathan** [comment]
Hello Tolusha,

Should I take 7.41.2 version signed off by yourself and try it out? Will Keycloak get installed by default in this version?

Spent too much time on version 7.42.0 for Keycloak and found the default installation got removed.

Should I enable devWorkspace to true or leave it to default false?

With Thanks,
Nathan CSA

**Comment by csa-nathan** [comment]
I tried 7.41.2 and its the same error. The websocket gets closed in between. Its something to do with CheClusterRoles.

All versions starting from 7.40.0 to 7.41.2 have the same error. 

Its quite frustrating. From version 7.42, OIDC Provider has been removed.   Installing Keycloak is another challenge. Initially i tried with 7.42 and could not succeed.

Kindly somebody let me know one version that would work after successful installation.  Create workspace and it should open up the Eclipse Che IDE.  Struggling for so many days on this activity...
